pykube/objects.py

```diff
@@ -161,18 +161,21 @@ def update(self, is_strategic=True, *, subresource=None):
         self.obj = obj_merge(self.obj, self._original_obj, is_strategic)
         self.patch(self.obj, subresource=subresource)
 
-    def delete(self, propagation_policy: str = None):
+    def delete(self, propagation_policy: str = None, extra_body=None, extra_params=None):
```

Comment:
Looking at the API ref it seems that everything that can be specified via DELETE request query params can also be specified via the request body, so I'd suggest removing the extra_params and their handling altogether.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#deleteoptions-v1-meta

Left are dryRun, gracePeriodSeconds and preconditions.
I'd suggest to put gracePeriodSeconds explicitly (as it is simple int) and for the rest (which are complex entities themselves) either ignore / do not support them at all for now.